autodist/kernel/graph_transformer.py

```diff
@@ -146,8 +146,10 @@ def _in_graph_apply(self, graph_item: GraphItem):
             GraphItem
         """
         new_graph_item = graph_item
+        new_graph_item.set_optimize()
```

Comment:
I don't quite understand why you need to call `set_optimize` then `reset_optimize`? could you explain a bit?

Reply:
I think the graph transformer needs to tell graph item when a loop of in_graph / between_graph is started or done. Current method in graph item manipulates a copy of all_update_op so it needs to know when to start the next loop. Also, it needs to turn on some flag when a new loop is started.